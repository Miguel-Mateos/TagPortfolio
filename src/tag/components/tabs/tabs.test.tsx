import {fireEvent, render} from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';
import Tabs, {ITabsProps, TabItem} from './tabs';

const tabsExample = (props?: ITabsProps) => {
  return (
    <Tabs {...props} className="tag-ds">
      <TabItem data-testid="tab1" id="tab1" title="Tab 1">
        <p>
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Consequatur saepe doloribus nisi quis consectetur numquam
          blanditiis accusamus dolor ipsam aspernatur aperiam aut accusantium, itaque quasi magni enim exercitationem
          repellat? Obcaecati. Lorem ipsum dolor sit amet, consectetur adipisicing elit. Lorem ipsum dolor sit amet,
          consectetur adipisicing elit.
        </p>
      </TabItem>
      <TabItem data-testid="tab2" id="tab2" title="Tab 2">
        <p>
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Minima, neque accusantium fugiat eveniet dignissimos quas!
          Rem cupiditate corrupti, molestias dolores distinctio excepturi perspiciatis numquam placeat ipsa necessitatibus
          facilis nobis optio quisquam, veritatis incidunt officiis at dolore accusantium porro sit, eaque qui atque quas
          suscipit. Voluptatibus asperiores quia dicta ducimus distinctio?
        </p>
      </TabItem>
      <TabItem collapsed data-testid="tab3" id="tab3" title="Tab 3">
        <p>
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Minima, neque accusantium fugiat eveniet dignissimos quas!
          Rem cupiditate corrupti, molestias dolores distinctio excepturi perspiciatis numquam placeat ipsa necessitatibus
          facilis nobis optio quisquam, veritatis incidunt officiis at dolore accusantium porro sit, eaque qui atque quas
          suscipit. Voluptatibus asperiores quia dicta ducimus distinctio? Lorem ipsum dolor sit amet consectetur adipisicing
          elit. Minima, neque accusantium fugiat eveniet dignissimos quas!
        </p>
      </TabItem>
      <TabItem collapsed data-testid="tab4" id="tab4" title="Tab 4">
        <p>
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Minima, neque accusantium fugiat eveniet dignissimos quas!
          Rem cupiditate corrupti, molestias dolores distinctio excepturi perspiciatis numquam placeat ipsa necessitatibus
          facilis nobis optio quisquam, veritatis incidunt officiis at dolore accusantium porro sit, eaque qui atque quas
          suscipit. Voluptatibus asperiores quia dicta ducimus distinctio?
        </p>
      </TabItem>
    </Tabs>
  );
};

const tabsExampleWithUniqueChild = (props?: ITabsProps) => {
  return (
    <Tabs {...props}>
      <TabItem data-testid="tab1" id="tab1" title="Tab 1">
        <p>
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Consequatur saepe doloribus nisi quis consectetur numquam
          blanditiis accusamus dolor ipsam aspernatur aperiam aut accusantium, itaque quasi magni enim exercitationem
          repellat? Obcaecati. Lorem ipsum dolor sit amet, consectetur adipisicing elit. Lorem ipsum dolor sit amet,
          consectetur adipisicing elit.
        </p>
      </TabItem>
    </Tabs>
  );
};

const tabsExampleWithUniqueChildCollapsed = (props?: ITabsProps) => {
  return (
    <Tabs {...props}>
      <TabItem collapsed data-testid="tab1" id="tab1" title="Tab 1">
        <p>
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Consequatur saepe doloribus nisi quis consectetur numquam
          blanditiis accusamus dolor ipsam aspernatur aperiam aut accusantium, itaque quasi magni enim exercitationem
          repellat? Obcaecati. Lorem ipsum dolor sit amet, consectetur adipisicing elit. Lorem ipsum dolor sit amet,
          consectetur adipisicing elit.
        </p>
      </TabItem>
    </Tabs>
  );
};

test('Tabs horizontal should render', () => {
  const component = renderer.create(tabsExample());
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Tabs horizontal with unique child should render', () => {
  const component = renderer.create(tabsExampleWithUniqueChild());
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Tabs horizontal with unique child collapsed should render', () => {
  const component = renderer.create(tabsExampleWithUniqueChildCollapsed());
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Tabs vertical should render', () => {
  const component = renderer.create(tabsExample({vertical: true}));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Select tab 2 ', () => {
  const {getByTestId} = render(tabsExample());

  const tabBtn2 = getByTestId('tab2');

  if (tabBtn2) fireEvent.click(tabBtn2);

  expect(getByTestId('tab2-content')).toBeDefined();
});

