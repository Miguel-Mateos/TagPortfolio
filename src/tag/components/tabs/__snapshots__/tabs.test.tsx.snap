// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs horizontal should render 1`] = `
<div
  className="tabs tag-ds"
>
  <div
    className="tab-list"
  >
    <button
      className="tab-list-item active"
      data-testid="tab1"
      id="tab1"
      onClick={[Function]}
    >
      Tab 1
    </button>
    <button
      className="tab-list-item "
      data-testid="tab2"
      id="tab2"
      onClick={[Function]}
    >
      Tab 2
    </button>
    <div
      className="dropdown "
    >
      <div
        className="dropdown-button "
        onClick={[Function]}
      >
        <button
          className=""
        >
          <span
            className="material-icons"
          >
            more_vert
          </span>
        </button>
      </div>
      <div
        className="dropdown-menu _divider  "
      >
        <button
          className="dropdown-item tab-list-item "
          data-testid="tab3"
          id="tab3"
          onClick={[Function]}
        >
          Tab 3
        </button>
        <button
          className="dropdown-item tab-list-item "
          data-testid="tab4"
          id="tab4"
          onClick={[Function]}
        >
          Tab 4
        </button>
      </div>
    </div>
  </div>
  <div
    className="tab-content"
    data-testid="tab1-content"
    id="tab1-content"
  >
    <p>
      Lorem ipsum dolor sit amet, consectetur adipisicing elit. Consequatur saepe doloribus nisi quis consectetur numquam blanditiis accusamus dolor ipsam aspernatur aperiam aut accusantium, itaque quasi magni enim exercitationem repellat? Obcaecati. Lorem ipsum dolor sit amet, consectetur adipisicing elit. Lorem ipsum dolor sit amet, consectetur adipisicing elit.
    </p>
  </div>
</div>
`;

exports[`Tabs horizontal with unique child collapsed should render 1`] = `
<div
  className="tabs "
>
  <div
    className="tab-list"
  />
</div>
`;

exports[`Tabs horizontal with unique child should render 1`] = `
<div
  className="tabs "
>
  <div
    className="tab-list"
  />
</div>
`;

exports[`Tabs vertical should render 1`] = `
<div
  className="tabs-vertical tag-ds"
>
  <div
    className="tab-list"
  >
    <button
      className="tab-list-item active"
      data-testid="tab1"
      id="tab1"
      onClick={[Function]}
    >
      Tab 1
    </button>
    <button
      className="tab-list-item "
      data-testid="tab2"
      id="tab2"
      onClick={[Function]}
    >
      Tab 2
    </button>
    <div
      className="dropdown "
    >
      <div
        className="dropdown-button "
        onClick={[Function]}
      >
        <button
          className=""
        >
          <span
            className="material-icons"
          >
            more_vert
          </span>
        </button>
      </div>
      <div
        className="dropdown-menu   "
      >
        <button
          className="dropdown-item tab-list-item "
          data-testid="tab3"
          id="tab3"
          onClick={[Function]}
        >
          Tab 3
        </button>
        <button
          className="dropdown-item tab-list-item "
          data-testid="tab4"
          id="tab4"
          onClick={[Function]}
        >
          Tab 4
        </button>
      </div>
    </div>
  </div>
  <div
    className="tab-content"
    data-testid="tab1-content"
    id="tab1-content"
  >
    <p>
      Lorem ipsum dolor sit amet, consectetur adipisicing elit. Consequatur saepe doloribus nisi quis consectetur numquam blanditiis accusamus dolor ipsam aspernatur aperiam aut accusantium, itaque quasi magni enim exercitationem repellat? Obcaecati. Lorem ipsum dolor sit amet, consectetur adipisicing elit. Lorem ipsum dolor sit amet, consectetur adipisicing elit.
    </p>
  </div>
</div>
`;
